code_search/docker/parse_arguments.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+# Common logic to parse the argument
+# To use it, define the usage() function for help message and name array variables for required arguments in the
+# parent script
+
+parseArgs() {
+  # Parse all command line options
+  while [[ $# -gt 0 ]]; do
+    # Parameters should be of the form
+    # --{name}=${value}
+    echo parsing "$1"
+    if [[ $1 =~ ^--(.*)=(.*)$ ]]; then
+    	name=${BASH_REMATCH[1]}
+    	value=${BASH_REMATCH[2]}
+     	eval ${name}="${value}"
+    elif [[ $1 =~ ^--(.*)$ ]]; then
+		name=${BASH_REMATCH[1]}
+		value=true
+		eval ${name}="${value}"
+    else
+    	echo "Argument $1 did not match the pattern --{name}={value} or --{name}"
+    fi
+    shift
+  done
+}
+
+parseArgs $*
```

Comment:
Did you mean to copy code below line 27 here? Or does this code belong in the main file?

Reply:
yeah that was intentional. It's a common logic so i did as much as possible to avoid duplication. let me know if you want to move them to parent script.